### PR TITLE
Add `PlacementInfo`

### PR DESCRIPTION
**Pull request by thevar1able**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow loading AZ info from a file.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit f678bdc546c400e32fb603c2e6e53e75ec2fa370 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59976/f678bdc546c400e32fb603c2e6e53e75ec2fa370/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59976/f678bdc546c400e32fb603c2e6e53e75ec2fa370/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Coordination/KeeperContext.cpp [hunk 1]
```diff
@@ -37,26 +38,11 @@ void KeeperContext::initialize(const Poco::Util::AbstractConfiguration & config,
 {
     dispatcher = dispatcher_;
 
-    if (config.hasProperty("keeper_server.availability_zone"))
+    const auto keeper_az = PlacementInfo::PlacementInfo::instance().getAvailabilityZone();
+    if (!keeper_az.empty())
     {
-        auto keeper_az = config.getString("keeper_server.availability_zone.value", "");
```
Comment (kssenii) [inline src/Coordination/KeeperContext.cpp:42]: should this old config path still be supported as a fallback for compatibility (at least for some period of time)?

Comment (thevar1able) [reply src/Coordination/KeeperContext.cpp:42]: I think no, I confirmed with @incfly that it is not used anywhere, and we did not document it.

**Review by kssenii — APPROVED** [review]

**Review by incfly — COMMENTED** [review]
Add some docs changes? for the configuration option.

File: src/Server/CloudPlacementInfo.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    LOG_DEBUG(log, "Loaded info: availability_zone: {}", availability_zone);
+    initialized = true;
+}
+
+std::string PlacementInfo::getAvailabilityZone() const
+{
+    if (!initialized)
+    {
+        LOG_WARNING(log, "Placement info has not been loaded");
+        return "";
```
Comment (tavplubix) [inline src/Server/CloudPlacementInfo.cpp:76]: Why just a warning? Maybe throw a LOGICAL_ERROR?

**Pull request merged**